src/classiceditorui.js

```diff
@@ -89,6 +104,35 @@ export default class ClassicEditorUI {
 				}
 
 				return Promise.all( promises );
+			} )
+			.then( () => {
+				const toolbarFocusTracker = this.view.toolbar.focusTracker;
+
+				// Because toolbar items can get focus, the overall state of
+				// the toolbar must also be tracked.
+				this.focusTracker.add( this.view.toolbar.element );
+
+				// Listen on the keystrokes from the main UI.
+				this.keystrokes.listenTo( this.view.element );
+
+				// Listen on the keystrokes from the floating panels, toolbars and the such.
+				this.keystrokes.listenTo( this.view._bodyCollectionContainer );
+
+				// Focus the toolbar on the keystroke, if not already focused.
+				this.keystrokes.set( 'alt + f10', ( data, cancel ) => {
+					if ( this.focusTracker.isFocused && !toolbarFocusTracker.isFocused ) {
+						this.view.toolbar.focus();
+						cancel();
+					}
+				} );
+
+				// Blur the toolbar and bring the focus back to editable on the keystroke.
+				this.keystrokes.set( 'esc', ( data, cancel ) => {
```

Comment:
> ToolbarView should never know about the environment it lives in except for what is passed in the constructor and what API it provides itself (observables).

But I didn't mean that it will have to know. The code will still be introduced by `ClassicEditorUI` but will use `ToolbarView`'s keystroke handler.

The goal is to remove unnecessary `ClassicEditorUI`'s keystroke handler. That's my point number 1.

Point number 2. is that your solution works differently than in CKEditor 4. The <kbd>Alt+F10</kbd> keystroke is handled on every part of the editor UI whereas, in CKEditor 4, it's only handled in editable. That, in my opinion, supports the claim that the additional keystroke handler should be removed.